staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -123,7 +123,7 @@ type DeltaFIFO struct {
 	// Used to indicate a queue is closed so a control loop can exit when a queue is empty.
 	// Currently, not used to gate any of CRED operations.
 	closed     bool
-	closedLock sync.Mutex
+	closedLock sync.RWMutex
```

Comment:
Did you profile this? This looks like a super dangerous change. We need to have a compelling reason to do it.